src/AutoRest.CSharp/LowLevel/Generation/LowLevelExampleComposer.cs

```diff
@@ -1124,7 +1125,9 @@ private static bool IsPropertyAssignable(ObjectTypeProperty property)
 
         private string FixKeyWords(string serializedName)
         {
-            var result = serializedName.Replace('-', '_');
+            // TODO -- this is incorrect, we should never change the property name here otherwise the serialized request content is wrong
+            // if the name changed after calling this method, we should use a dictionary instead of using anonymous object
+            var result = serializedName.Replace('-', '_').Replace(".", string.Empty);
```

Comment:
Each property has `name` and `serializedName`. `Name` will be the variable name in the model which should be updated to comply c# naming convention. `serializedName` is used in serialization which should not be updated. In this function, the input parameter string is not updated. That's why it does not bring error during serialization. 
Maybe we need to update the method signature `private string FixKeyWords(string name)` to avoid confusing, and we need call it for property name.

Reply:
This is solving the case that serializedName has special characters. We could have a serialized name called "string", or "name.kind" which are both invalid identifier.
This is not about something is confusing, I added this todo and change the code to temporary workaround an issue I found in the samples.
To make things correct, we should never change anything in the "serialized name" as my statement here which means we should remove this method in the future.
I already opened an issue for this: https://github.com/Azure/autorest.csharp/issues/3499